common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/storage/BlockConnectionStorage.java

```diff
@@ -118,11 +123,15 @@ public void unloadChunk(int x, int z) {
     }
 
     public void unloadSection(int x, int y, int z) {
-        blockStorage.remove(getChunkSectionIndex(x << 4, y << 4, z << 4));
+        long index = getChunkSectionIndex(x << 4, y << 4, z << 4);
+        blockStorage.remove(index);
+        if (index == lastIndex) {
+            lastSection = null;
+        }
     }
 
     private SectionData getChunkSection(long index, boolean requireNibbleArray) {
-        SectionData map = blockStorage.get(index);
+        SectionData map = getSection(index);
         if (map == null) {
             map = new SectionData(new byte[4096]);
             blockStorage.put(index, map);
```

Comment:
Could probably also set the cached value here, assuming it's going to be called multiple times after computed